### PR TITLE
Cross Build for armv7

### LINKED ISSUE #440: cross compilation failing for aarch64 target in github actions
e.g. https://github.com/andrewdavidmackenzie/pigg/actions/runs/10850493649

I might disable cross build, so CI is green, until we can fix this...

### PR DESCRIPTION
**Pull request by sundaram123krishnan**
Cross compile for armv7

### CODE REVIEW

**Comment by sundaram123krishnan** [comment]
When changed the target architecture to `armv7-unknown-linux-gnueabihf` it works fine @andrewdavidmackenzie

**Comment by sundaram123krishnan** [comment]
At first we used to test our cross compilation with only `aarch64-unknown-linux-gnu`

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -31,27 +31,32 @@ jobs:
...
+    steps:
+      - name: Clean up files
+        run: |
+          sudo rm -rf /usr/share/dotnet
+          sudo rm -rf /opt/ghc
+          sudo rm -rf "/usr/local/share/boost"
+          sudo rm -rf "$AGENT_TOOLSDIRECTORY"
+
+#      - name: Install Linker
+#        run: |
```
Comment (andrewdavidmackenzie) [inline .github/workflows/build_and_test.yml:45]: We can leave the install of the other target, as we will need to build for both….

File: Makefile [hunk 2]
```diff
@@ -19,7 +19,7 @@ $(eval PI = $(shell cat /proc/cpuinfo 2>&1 | grep "Raspberry Pi"))
 all: clippy build test
 
 .PHONY: cross-ci
-cross-ci: cross-clippy cross-build-aarch64 cross-test-aarch64
+cross-ci: cross-clippy cross-build-armv7 cross-test-armv7
```
Comment (andrewdavidmackenzie) [inline Makefile:22]: Let’s change it so it does cross for both arch?
Either that or two make targets (one for each arch), and have two GH Action jobs, one for arm7 and one for aarch64 - but will be slightly quicker if same job does both (only one install of cross etc)

**Review by andrewdavidmackenzie — COMMENTED** [review]
So, it’s great that armv7 works - but we really need aarch64 also for newer Pis and 64bit…

Options
1) remove the “Closes” from the comment and merge this, and leave issue open with comment about aarch64 needing fixing
2) merge this as-is, and create a new issue for aarch64 also
3) Not merge and try and get aarch64 working…

**Pull request merged**